android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java

```diff
@@ -8,20 +9,77 @@
 import com.facebook.common.logging.FLog;
 import com.facebook.react.uimanager.ThemedReactContext;
 
+import java.util.Map;
+
 public class RNCWebView extends FrameLayout {
   private static final String TAG = "RNCWebView";
   public static final int INVALID_VIEW_ID = -1;
-
+  private String webViewKey;
   private RNCWebViewManager.InternalWebView internalWebView;
 
   public RNCWebView(ThemedReactContext reactContext) {
     super(reactContext);
+
+    // There is an issue with react-native where if a view is moved
+    // the parent and children views are not resized
+    // Since we are moving views outside of React, the layout request might be dropped
+    // on a normal add/remove view
+    // By calling .layout directly on the parent, it'll force the layout change
+    // See this issue for more context: https://github.com/facebook/react-native/issues/17968
+    setOnHierarchyChangeListener(new OnHierarchyChangeListener() {
+      @Override
+      public void onChildViewAdded(View parent, View child) {
+        if (parent != null) {
+          parent.measure(
+            MeasureSpec.makeMeasureSpec(getMeasuredWidth(), MeasureSpec.EXACTLY),
+            MeasureSpec.makeMeasureSpec(getMeasuredHeight(), MeasureSpec.EXACTLY)
+          );
+          parent.layout(0, 0, parent.getMeasuredWidth(), parent.getMeasuredHeight());
```

Comment:
nit: It may be slightly more readable to name the l / t / r / b parameters inline with code comments sort of like this.

```
parent.layout(
  0 /* left */ ,
  0 /* top */,
  parent.getMeasuredWidth() /* right */,
  parent.getMeasuredHeight() /* bottom */
);
```